src/node_contextify.cc

```diff
@@ -1339,6 +1341,104 @@ Local<Object> ContextifyContext::CompileFunctionAndCacheResult(
   return result;
 }
 
+// These are the error messages thrown due to ESM syntax in a CommonJS module.
+constexpr std::array<std::string_view, 3> esm_syntax_error_messages = {
+    // `import` statements return an error with the message:
+    "Cannot use import statement outside a module",
+    // `export` statements return an error with the message:
+    "Unexpected token 'export'",
+    // `import.meta` returns an error with the message:
+    "Cannot use 'import.meta' outside a module"};
+// Top-level `await` currently returns the same error message as when `await` is
+// used in a sync function, so we don't use it as a disambiguation. Dynamic
+// `import()` is permitted in CommonJS, so we don't use it as a disambiguation.
```

Comment:
These comments reduces the readability. Can you move them the line before constexpr declaration?

Reply:
They're here because I wanted to explain the errors we look for first before mentioning a few others we ignore.